SparseGrids/tsgRuleLocalPolynomial.hpp

```diff
@@ -364,6 +463,79 @@ class templRuleLocalPolynomial : public BaseRuleLocalPolynomial{
         }
         return value;
     }
+
+    // Same logic as the corresponding eval functions.
+    double diffPWQuadratic(int point, double x) const{
+        if (rule == rule_localp) {
+            if (point == 1) return -1.0;
+            if (point == 2) return 1.0;
+        } else if (rule == rule_localpb) {
+            if (point == 0) return -1.0;
+            if (point == 1) return 1.0;
+        }
+        return -2.0 * x;
+    }
+    double diffPWCubic(int point, double x) const{
+        if (rule == rule_localp) {
+            if (point == 0) return 0.0;
+            if (point == 1) return -1.0;
+            if (point == 2) return 1.0;
+            if (point <= 4) return -2.0 * x;
+        } else if (rule == rule_localpb) {
+            if (point == 0) return -1.0;
+            if (point == 1) return 1.0;
+            if (point == 2) return -2.0 * x;
+        } else if (rule == rule_localp0) {
+            if (point == 0) return -2.0 * x;
+        }
+        return (point % 2 == 0) ? 1.0 / 3.0 - x * (x + 2.0) : -1.0 / 3.0 + x * (x - 2.0);
+    }
+    double diffPWPower(int point, double x) const{
+        if (rule == rule_localp)     if (point <= 8) return diffPWCubic(point, x);
```

Comment:
Why nest two if-statements when you can just use `and` logic.

Reply:
Changed. This was copied from `evalPWPower()`, and I'm always hesitant to modify the structure of existing code because I don't know if there is some hidden logic that I missed.

Should I also change other code segments where I see nested if-statements like this (e.g. `evalPWPower()` and `evalRaw()`)?